pkg/bspatch/bspatch.go

```diff
@@ -59,34 +67,51 @@ func Reader(oldbin io.Reader, newbin io.Writer, patchf io.Reader) error {
 
 // File applies a BSDIFF4 patch (using oldfile and patchfile) to create the newfile
 func File(oldfile, newfile, patchfile string) error {
-	oldbs, err := ioutil.ReadFile(oldfile)
+	oldf, err := os.Open(oldfile)
+	if err != nil {
+		return fmt.Errorf("could not open oldfile '%v': %v", oldfile, err.Error())
+	}
+	defer oldf.Close()
+
+	newf, err := os.Create(newfile)
 	if err != nil {
-		return fmt.Errorf("could not read oldfile '%v': %v", oldfile, err.Error())
+		return fmt.Errorf("could not open or create newfile '%v': %v", newfile, err.Error())
```

Comment:
- use `%s` for strings
- pass `err` directly – `fmt` knows how to handle errors